src/AppBundle/Repository/EventRepository.php

```diff
@@ -236,14 +263,51 @@ private function getFileUsageBuilder(string $dbName, DateTime $start, DateTime $
         }
 
         $rqb->andWhere('img_timestamp BETWEEN :start AND :end')
-            ->andWhere('img_user_text IN (:usernames)')
             ->setParameter('start', $start)
             ->setParameter('end', $end)
             ->setParameter('usernames', $usernames, Connection::PARAM_STR_ARRAY);
 
+        $this->applyUsersOrCategories($rqb, $dbName, $usernames, $categories);
+
         return $rqb;
     }
 
+    /**
+     * @param QueryBuilder $rqb
+     * @param string $dbName
+     * @param string[] $usernames
+     * @param string[] $categories
+     */
+    private function applyUsersOrCategories(
+        QueryBuilder $rqb,
+        string $dbName,
+        array $usernames,
+        array $categories
+    ): void {
+        if ($usernames && $categories) {
+            // throw new InvalidParameterException('Cannot process events with both users and categories');
+        }
+        if (!$usernames && !$categories) {
+            throw new InvalidParameterException('Events need either users or categories');
+        }
+
+        $namespace = 'commonswiki_p' === $dbName ? 6 /* NS_FILE */ : 0 /* NS_MAIN */;
+        if ($usernames) {
+            $rqb->andWhere("img_user_text IN (:usernames)")
+                ->setParameter('usernames', $usernames, Connection::PARAM_STR_ARRAY);
+        } elseif ($categories) {
+            $rqb->join(
+                "$dbName.image",
+                "$dbName.page",
+                null,
+                "img_name = page_title AND page_namespace = /* NS_FILE */ $namespace"
```

Comment:
Guessing you didn't mean for `/* NS_FILE */` to be here, since it's not necessarily the file namespace

Reply:
Done.